components/definitions/newTab.d.ts

```diff
@@ -40,25 +40,32 @@ declare namespace NewTab {
     url: string
   }
 
-  export interface State {
-    textDirection: string
+  export interface PersistantState {
     topSites: Site[]
     ignoredTopSites: Site[]
     pinnedTopSites: Site[]
     gridSites: Site[]
     showEmptyPage: boolean
+    bookmarks: Record<string, Bookmark>
+  }
+
+  export interface EphemeralState {
+    initialDataLoaded: boolean
+    textDirection: string
     isIncognito: boolean
     useAlternativePrivateSearchEngine: boolean
     isTor: boolean
     isQwant: boolean
-    bookmarks: Record<string, Bookmark>
-    stats: Stats
     backgroundImage?: Image
```

Comment:
out of context but just realized we didn't propagate `Image | undefined` down to the actual components using this data - can be done in another PR but quick change if you want to take it 😄

Reply:
@imptrx Can you give me an example? All the places I can see do do a type check first. Typescript should produce a compiler error if they didn't.